### PR TITLE
Add Login Logic. Add target to logout button on HomeViewController. C…

### PR DESCRIPTION
**Pull request by tamokakhidze**
…hange initial viewController setup in SceneDelegate.

### CODE REVIEW

File: QuizApp/CoreData/LoginManager.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+//  LoginManager.swift
+//  QuizApp
+//
+//  Created by Tamuna Kakhidze on 27.11.24.
+//
+
+import CoreData
+import UIKit
+
+class LoginManager {
```
Comment (AnaIoramashvili) [inline QuizApp/CoreData/LoginManager.swift]: please add final keyword

**Review by AnaIoramashvili — CHANGES_REQUESTED** [review]

File: QuizApp/CoreData/LoginManager.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    
+    // MARK: - Get Current User's Data
+    func getCurrentUserName() -> String? {
+        return currentUser?.value(forKey: "name") as? String
+    }
+    
+    func getCurrentUserPoints() -> [String: Int]? {
+        return currentUser?.value(forKey: "points") as? [String: Int]
+    }
+
```
Comment (AnaIoramashvili) [inline QuizApp/CoreData/LoginManager.swift]: remove whitespaces

**Review by AnaIoramashvili — APPROVED** [review]

File: QuizApp/Scenes/LoginPage/ViewModel/LoginViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+//
+//  Created by Tamuna Kakhidze on 27.11.24.
+//
+
+import Foundation
+
+class LoginViewModel {
+    
+    // MARK: - Properties
+    let loginManager = LoginManager.shared
```
Comment (LukaGujejiani) [inline QuizApp/Scenes/LoginPage/ViewModel/LoginViewModel.swift]:
```suggestion
    private let loginManager = LoginManager.shared
```

File: QuizApp/Scenes/LoginPage/ViewModel/LoginViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
+//
+//  LoginViewModel.swift
+//  QuizApp
+//
+//  Created by Tamuna Kakhidze on 27.11.24.
+//
+
+import Foundation
+
+class LoginViewModel {
```
Comment (LukaGujejiani) [inline QuizApp/Scenes/LoginPage/ViewModel/LoginViewModel.swift]:
```suggestion
final class LoginViewModel {
```

**Review by LukaGujejiani — APPROVED** [review]

**Pull request merged**